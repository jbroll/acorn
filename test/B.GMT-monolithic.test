#!/usr/bin/env tclkit8.6
#
lappend auto_path ./lib ../lib

package require acorn

acorn::surface-group create GMT {
    .		{ 								thickness -23000      		}
    M2Objc	{ 								thickness  23000      		}
    Primary	{ aper 0	R -36000.0	K  -0.99828570	n  -1.00	thickness -20262.476419940002	}
    Secondary	{ aper 0	R   4163.900922 K  -0.71692784	n  -1.00	thickness  20262.476419940002	}
    .		{ 								thickness   4900      		}
    .		{        							thickness    620      		}
    .   	{ 								thickness    310      		}
    Fous	{ 								thickness      0      		}
}



#acorn::Rays new rays 1
#rays 0 set px -12724.0 py -12724.0 pz 0 kx 0.0 ky 0.0 kz 1.0
#acorn::trace rays GMT
#puts "x     	y     	z      	kx	ky	kz"
#puts "-------	-------	-------	--	--	--"
#puts [join [map row [rays 0 end get px py pz kx ky kz] { join [map x $row { format "% .7f" $x }] \t }] \n]
#exit



acorn::Rays new rays 9

set i 0
foreach x { -12700 0 12700 } {
    foreach y { -12700 0 12700 } {
	rays $i set px $x py $y pz 0 kx 0.0 ky 0.0 kz 1.0
	incr i
    }
}

acorn::trace rays GMT


puts "x     	y     	z      "
puts "-------	-------	-------"
puts [join [map row [rays 0 end get px py pz] { join [map x $row { format "% .4f" $x }] \t }] \n]


