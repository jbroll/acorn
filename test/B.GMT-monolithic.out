x     	y     	z      
-------	-------	-------
 0.4492	 0.4492	 23520.0000
 0.4492	 0.0000	 23520.0000
 0.4492	-0.4492	 23520.0000
 0.0000	 0.4492	 23520.0000
 0.0000	 0.0000	 23520.0000
 0.0000	-0.4492	 23520.0000
-0.4492	 0.4492	 23520.0000
-0.4492	 0.0000	 23520.0000
-0.4492	-0.4492	 23520.0000
